CONTRIBUTING.md

```diff
@@ -1,103 +1,155 @@
 
 ```shell
-pip install --upgrade wandb
+./tools/setup_dev_environment.py
 ```
 
-## Code organization
+At the first invocation, this tool will set up multiple python environments, which takes some time.
+You can set up a subset of the target environments to test against, for example:
 
-```bash
-wandb/
-├── ...
-├── apis/   # Public api (still has internal api but this should be moved to wandb/internal)
-│   ├── ...
-│   ├── internal.py
-│   ├── ...
-│   └── public.py
-├── cli/    # Handlers for command line functionality
-├── ...
-├── integration/    # Third party integration
-│   ├── fastai/
-│   ├── gym/
-│   ├── keras/
-│   ├── lightgbm/
-│   ├── metaflow/
-│   ├── prodigy/
-│   ├── sacred/
-│   ├── sagemaker/
-│   ├── sb3/
-│   ├── tensorboard/
-│   ├── tensorflow/
-│   ├── torch/
-│   ├── xgboost/
-│   └── ...
-├── ...
-├── proto/  # Protocol buffers for inter-process communication and persist file store
-├── ...
-├── sdk/    # User accessed functions [wandb.init()] and objects [WandbRun, WandbConfig, WandbSummary, WandbSettings]
-│   ├── backend/    # Support to launch internal process
-│   ├── ...
-│   ├── interface/  # Interface to backend execution
-│   ├── internal/   # Backend threads/processes
-│   └── ...
-├── ...
-├── sweeps/ # Hyperparameter sweep engine (see repo: https://github.com/wandb/sweeps)
-└── ...
+```shell
+./tools/setup_dev_environment.py -p 3.7 3.8
 ```
 
-## Setup development environment
-
-In order to run unittests please install pyenv:
+The tool will also set up [`tox`](https://github.com/tox-dev/tox), which we use 
+for automating development tasks such as code linting and testing. 
 
-(put the output in your `~/.bashrc`)
+## Building protocol buffers
 
-```shell
-curl https://pyenv.run | bash
-```
+We use [protocol buffers](https://developers.google.com/protocol-buffers) to communicate 
+from the user process to the `wandb` backend process.
 
-then run:
+If you update any of the `.proto` files in `wandb/proto`, you'll need to run:
 
 ```shell
-./tools/setup_dev_environment.sh
+make proto
 ```
 
-## Building protocol buffers
-
-We use protocol buffers to communicate from the user process to the wandb backend process.
+## Linting the code
 
-If you update any of the .proto files in `wandb/proto`, you'll need to run:
+We use [`black`](https://black.readthedocs.io/), [`flake8`](https://flake8.pycqa.org/), 
+and [`mypy`](http://mypy-lang.org/) for code formatting and checks (including static type checks). 
 
+To reformat the code, run: 
 ```shell
-make proto
+tox -e format
 ```
 
-## Code checks
+To run checks, execute: 
 
-- Reformat: `tox -e format`
-- Type check: `tox -e flake8,mypy`
-- Misc: `tox`
+```shell
+tox -e flake8,mypy
+```
 
 ## Testing
 
-Tests can be found in `tests/`. We use tox to run tests, you can run all tests with:
+We use the [`pytest`](https://docs.pytest.org/) framework. Tests can be found in `tests/`. You can run all tests with:
 
 ```shell
 tox
```

Comment:
we should probably not recommend running `tox` without any other args because this will run so many things, and may spit out unfriendly errors if the users dont have the proper python versions installed. we should just defer to one like `py37`

Reply:
👍🏻